### PR TITLE
[dv] Use platform-agnostic logs everywhere applicable

### PR DESCRIPTION
**Pull request by sriyerg**
- Added `` `DV_INFO|WARNING|ERROR|FATAL`` that handles the switch between UVM logging and standard SV logging. 
- Updated all existing modules to use these macros instead of adding `` `ifdef VERILATOR|UVM`` everywhere
  - `prim_assert`, `sw_test_status`, `sw_logger_if`

### CODE REVIEW

File: hw/dv/sv/sw_test_status/sw_test_status_if.sv [hunk 1]
```diff
@@ -16,6 +16,9 @@ interface sw_test_status_if #(
 `endif
   import sw_test_status_pkg::*;
 
+  // macro includes
+  `include "dv_macros.svh"
+
```
Comment (rswarbrick) [inline hw/dv/sv/sw_test_status/sw_test_status_if.sv:21]: Nit: is this supposed to be in a different commit, maybe?

**Review by rswarbrick — APPROVED** [review]
Looks good to me!

**Review by imphil — COMMENTED** [review]
This change means that effectively all RTL code produced in OpenTitan or related to OpenTitan (such as Ibex) requires `dv_cores.svh` (well, at least all RTL using asserts). `dv_macros.svh` is a 15 kB source file with a lot of rather generically named symbols/macros.

For example, all users of Ibex now need to include `dv_utils.svh` as well (or need to provide similar macros and modify the source code accordingly). If they include the whole `dv_utils.svh` file, they need to be aware of name clashes, e.g. if they already have a `DV_CHECK_GE` function defined which  might be implemented differently.

I find this a rather large dependency to add for just an error reporting macro. As a way out, can we provide a minimal version of `DV_ERROR` that uses `$error` within `prim_assert_standard_macros.svh` if `DV_ERROR` is not defined? That still doesn't solve the problem of how to specify the conditional dependency in fusesoc, but maybe we can push that discussion out further. But ideally, we can find a solution which doesn't need the dependency at all.

File: hw/dv/sv/dv_utils/dv_macros.svh [hunk 2]
```diff
@@ -400,13 +407,15 @@
 // testbenches.
 `ifdef UVM
 `ifndef DV_INFO
-  `define DV_INFO(_msg, _id = $sformatf("%m"), _verbosity = UVM_LOW) \
-    `uvm_info(_id, _msg, _verbosity)
+  `define DV_INFO(MSG_,  VERBOSITY_ = uvm_pkg::UVM_LOW, ID_ = $sformatf("%m")) \
+    if (uvm_pkg::uvm_report_enabled(VERBOSITY_, uvm_pkg::UVM_INFO, ID_)) begin \
+        uvm_pkg::uvm_report_info(ID_, MSG_, VERBOSITY_, `uvm_file, `uvm_line, "", 1); \
+    end
```
Comment (weicaiyang) [inline hw/dv/sv/dv_utils/dv_macros.svh:413]: not sure why we don't just use `uvm_info

Comment (sriyerg) [reply hw/dv/sv/dv_utils/dv_macros.svh:413]: Great question! I should add a comment for this. 
Notice the difference between `uvm_info expansion and this one - all UVM specific things are prefixed with `uvm_pkg::`, which removes the need to `import uvm_pkg::*` in all the RTL modules (since this is used in `prim_assert.sv`).

File: hw/dv/sv/sw_test_status/sw_test_status_if.sv (hunk 1 above)
Comment (sriyerg) [reply hw/dv/sv/sw_test_status/sw_test_status_if.sv:21]: A separate commit will cause Verilator compile to fail on this commit.

File: hw/dv/sv/dv_utils/dv_macros.svh (hunk 2 above)
Comment (weicaiyang) [reply hw/dv/sv/dv_utils/dv_macros.svh:413]: I see. Thanks for the explanation. If we decide to remove the dependency of dv_util in RTL (define its own error report in prim_assert), we can switch back to uvm_info

**Comment by sriyerg** [comment]
> This change means that effectively all RTL code produced in OpenTitan or related to OpenTitan (such as Ibex) requires `dv_cores.svh` (well, at least all RTL using asserts). `dv_macros.svh` is a 15 kB source file with a lot of rather generically named symbols/macros.
> 
> For example, all users of Ibex now need to include `dv_utils.svh` as well (or need to provide similar macros and modify the source code accordingly). If they include the whole `dv_utils.svh` file, they need to be aware of name clashes, e.g. if they already have a `DV_CHECK_GE` function defined which might be implemented differently.
> 
> I find this a rather large dependency to add for just an error reporting macro. As a way out, can we provide a minimal version of `DV_ERROR` that uses `$error` within `prim_assert_standard_macros.svh` if `DV_ERROR` is not defined? That still doesn't solve the problem of how to specify the conditional dependency in fusesoc, but maybe we can push that discussion out further. But ideally, we can find a solution which doesn't need the dependency at all.

Very well, I can just replicate the `DV_ERROR macro in prim_assert. Macro redefinition shouldn't be a problem because of the ifndef guard.

**Review by weicaiyang — APPROVED** [review]
LGTM

**Review by imphil — APPROVED** [review]
LGTM, thanks for the update Sri!

File: hw/dv/sv/sw_test_status/sw_test_status_if.sv (hunk 1 above)
Comment (rswarbrick) [reply hw/dv/sv/sw_test_status/sw_test_status_if.sv:21]: Ah, I see. Thanks for the explanation.
